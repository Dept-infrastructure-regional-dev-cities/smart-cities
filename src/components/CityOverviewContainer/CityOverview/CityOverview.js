import React from 'react';
import PropTypes from 'prop-types';
import PageWrapper from '../../PageWrapper/PageWrapper';
import style from './CityOverview.scss';

const CityOverview = (props) => {
  const title = `I am the overview page for "${props.city.name}"`;

  return (
    <PageWrapper cityId={props.city.id}>
      <div className={style.main}>
        <h1>{title}</h1>

        <pre>{JSON.stringify(props.city, null, 2)}</pre>
      </div>
    </PageWrapper>
  );
};

CityOverview.propTypes = {
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  }).isRequired,
};

export default CityOverview;
