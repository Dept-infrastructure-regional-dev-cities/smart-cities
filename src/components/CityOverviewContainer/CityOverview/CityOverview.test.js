import React from 'react';
import { shallow } from 'enzyme';
import CityOverview from './CityOverview';

const defaultProps = {
  city: {
    name: 'Perth',
    id: 'perth',
  },
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityOverview {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
