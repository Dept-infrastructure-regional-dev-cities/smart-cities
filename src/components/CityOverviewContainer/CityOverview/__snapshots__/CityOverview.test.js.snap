// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper cityId=\\"perth\\" categoryId=\\"overview\\">
  <div className=\\"main\\">
    <h1>
      I am the overview page for &quot;Perth&quot;
    </h1>
    <pre>
      {
        &quot;name&quot;: &quot;Perth&quot;,
        &quot;id&quot;: &quot;perth&quot;
      }
    </pre>
  </div>
</PageWrapper>"
`;
